src/prevector.h

```diff
@@ -147,14 +146,20 @@ class prevector {
     };
 
 private:
-    size_type _size = 0;
+#pragma pack(push, 1)
     union direct_or_indirect {
         char direct[sizeof(T) * N];
         struct {
-            size_type capacity;
             char* indirect;
+            size_type capacity;
         };
-    } _union = {};
+    };
+#pragma pack(pop)
+    alignas(char*) direct_or_indirect _union = {};
+    size_type _size = 0;
+
+    static_assert(alignof(char*) % alignof(size_type) == 0 && sizeof(char*) % alignof(size_type) == 0, "size_type cannot have more restrictive alignment requirement than pointer");
```

Comment:
nit: `assert(A && B);` is equivalent to `assert(A); assert(B);`, however the former has the (small) problem that if it fails it is not clear from the message which one of `A` or `B` is false.

Reply:
`sizeof(X)` is always a multiple of `alignof(X)`, so I think the second half here is just redundant.